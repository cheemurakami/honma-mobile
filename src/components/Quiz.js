import * as a from "../rdx/actions";

import React, { useState } from "react";

import { Alert } from "react-native";
import { Button } from "react-native-paper";
import { TextInput } from "react-native-paper";
import { connect } from "react-redux";
import styled from "styled-components/native";

export const Quiz = ({ selectedDialect, grammar, auth, dispatch }) => {
  const [text, setText] = useState("");
  const [showCorrectButton, setShowCorrectButton] = useState(false);

  if (grammar.quizzes.length > 0) {
    const quiz = grammar.quizzes[0];
    const quizTokyo = quiz.tokyo;
    const answer = quiz.answer;

    const checkAnswer = (text) => {
      if (text === answer) {
        const action = a.completedGrammars(grammar.id);
        dispatch(action);
        const data = {
          quiz_id: quiz.id,
          authentication_token: auth.auth_token,
        };
        completedQuiz(data);
        setShowCorrectButton(true);
      } else {
        setShowCorrectButton(false);
        return Alert.alert("不正解！もう1度！");
      }
    };

    const completedQuiz = (data) => {
      fetch("http://localhost:3000/api/quiz_completions", {
        method: "POST",
        headers: {
          Accept: "application/json",
          "Content-Type": "application/json",
        },
        body: JSON.stringify(data),
      })
        .then((resp) => resp.json())
        .then((resp) => console.log(resp));
    };

    return (
      <>
        <BodyText>Please write this in {selectedDialect.name_en}:</BodyText>
        <BodyText>{quizTokyo}</BodyText>
        <TextInput
          label="Answer here"
          value={text}
          onChangeText={(text) => setText(text)}
          style={{ margin: 20 }}
        ></TextInput>
        <ButtonContainer>
          <Button
            mode="contained"
            onPress={() => {
              checkAnswer(text);
            }}
            color={showCorrectButton ? "#BDB2FF" : "#40BA62"}
            labelStyle={{ color: "#fff", fontSize: 18 }}
            style={{
              width: "50%",
              height: 45,
              justifyContent: "center",
            }}
          >
            {showCorrectButton ? "正解！！" : "答え合わせ"}
          </Button>
        </ButtonContainer>
      </>
    );
  } else {
    return <BodyText>More quizzes coming soon!</BodyText>;
  }
};

const BodyText = styled.Text`
  text-align: center;
  font-size: 18px;
  margin: 8px;
  font-weight: bold;
`;

const ButtonContainer = styled.View`
  align-items: center;
  margin: 10px;
`;

const mapStateToProps = (state) => {
  return {
    auth: state.authReducer,
  };
};

export default connect(mapStateToProps)(Quiz);
