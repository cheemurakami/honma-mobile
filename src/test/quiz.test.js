import React from "react";
import renderer from "react-test-renderer";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import rootReducer from "../rdx/reducers/index.js";
import Quiz from "../components/Quiz.js";
import {
  waitFor,
  render,
  screen,
  userEvent,
} from "@testing-library/react-native";
import {
  dialectData,
  grammarWithOneQuiz,
  grammarWithMultipleQuiz,
} from "./test-data/quiz-test-data.js";

describe("Quiz tests", () => {
  const store = configureStore({ reducer: rootReducer });
  const auth = {
    auth_token: "asdc",
    email: "test@test.com",
  };
  const mockDispatch = jest.fn();

  const renderComponent = (quizCount) => {
    const grammar =
      quizCount === "one quiz" ? grammarWithOneQuiz : grammarWithMultipleQuiz;

    return render(
      <Provider store={store}>
        <Quiz
          selectedDialect={dialectData}
          grammar={grammar}
          auth={auth}
          dispatch={mockDispatch}
        />
      </Provider>
    );
  };

  it("matches snapshot", async () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Quiz
            selectedDialect={dialectData}
            grammar={grammarWithOneQuiz}
            auth={auth}
            dispatch={mockDispatch}
          />
        </Provider>
      )
      .toJSON();
    await waitFor(() => {
      expect(tree).toMatchSnapshot();
    });
  });

  it("renders Quiz component properly", async () => {
    renderComponent("one quiz");

    const questionHeaderText = screen.getByText(
      `Please write this in ${dialectData.name_en}:`
    );
    const questionText = screen.getByText(
      `Q1: ${grammarWithOneQuiz.quizzes[0].tokyo}`
    );
    const textInput = screen.getByTestId("text-input-flat");
    const btn = screen.getByRole("button", { name: "答え合わせ" });

    await waitFor(() => {
      expect(questionHeaderText).toBeTruthy();
      expect(questionText).toBeTruthy();
      expect(questionText).toBeTruthy();
      expect(textInput).toBeTruthy();
      expect(btn).toBeTruthy();
    });
  });

  it("pops up an alert when input answer is incorrect", async () => {
    renderComponent("one quiz");

    const textInput = screen.getByTestId("text-input-flat");
    await userEvent.press(textInput);
    await userEvent.type(textInput, "asdc");
    const incorrectMessage = "不正解！もう１度！";

    await waitFor(() => {
      expect(incorrectMessage).toBeTruthy();
    });
  });

  it("shows correct button text with one quiz or the last quiz completion", async () => {
    renderComponent("one quiz");

    const textInput = screen.getByTestId("text-input-flat");
    await userEvent.press(textInput);
    await userEvent.type(textInput, grammarWithOneQuiz.quizzes[0].answer);

    const button = screen.getByRole("button", { name: "答え合わせ" });
    await userEvent.press(button);

    await waitFor(() => {
      const correctButton = screen.getByRole("button", {
        name: "全問正解！",
      });
      expect(correctButton).toBeTruthy();
    });
  });

  it("shows correct button text with multiple quizzes", async () => {
    renderComponent("multiple quizzes");

    const textInput = screen.getByTestId("text-input-flat");
    await userEvent.press(textInput);
    await userEvent.type(textInput, grammarWithMultipleQuiz.quizzes[0].answer);

    const button = screen.getByRole("button", { name: "答え合わせ" });
    await userEvent.press(button);

    await waitFor(() => {
      const correctButton = screen.getByRole("button", {
        name: "正解！次の問題へ",
      });
      expect(correctButton).toBeTruthy();
    });
  });

  it("renders next quiz after getting the correct answer when grammar has multiple quizzes", async () => {
    renderComponent("multiple quiz");

    const textInput = screen.getByTestId("text-input-flat");
    await userEvent.press(textInput);
    await userEvent.type(textInput, grammarWithMultipleQuiz.quizzes[0].answer);

    const button = screen.getByRole("button", { name: "答え合わせ" });
    await userEvent.press(button);

    await waitFor(() => {
      const correctButton = screen.getByRole("button", {
        name: "正解！次の問題へ",
      });
      expect(correctButton).toBeTruthy();
      userEvent.press(correctButton);
    });

    await waitFor(() => {
      const newQuestionText = screen.getByText(
        `Q2: ${grammarWithMultipleQuiz.quizzes[1].tokyo}`
      );

      const answerBtn = screen.getByRole("button", {
        name: "答え合わせ",
      });
      expect(newQuestionText).toBeTruthy();
      expect(answerBtn).toBeTruthy();
    });
  });
});
