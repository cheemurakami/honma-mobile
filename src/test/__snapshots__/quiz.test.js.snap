// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz tests matches snapshot 1`] = `
[
  <Text
    style={
      [
        {
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 8,
          "marginLeft": 8,
          "marginRight": 8,
          "marginTop": 8,
          "textAlign": "center",
        },
      ]
    }
  >
    Please write this in 
    Hiroshima
    :
  </Text>,
  <Text
    style={
      [
        {
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 8,
          "marginLeft": 8,
          "marginRight": 8,
          "marginTop": 8,
          "textAlign": "center",
        },
      ]
    }
  >
    Q1
    : 
    あの子は私の知り合いなんだよ。
  </Text>,
  <View
    style={
      [
        {
          "backgroundColor": "rgb(231, 231, 231)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        {
          "margin": 20,
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.26)",
          "bottom": 0,
          "height": 2,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": [
            {
              "scaleY": 0.5,
            },
          ],
          "zIndex": 1,
        }
      }
    />
    <View
      style={
        [
          {
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          {
            "minHeight": 64,
          },
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="none"
        style={
          {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": [
              {
                "translateX": 0,
              },
            ],
            "zIndex": 3,
          }
        }
      >
        <Text
          collapsable={false}
          maxFontSizeMultiplier={1.5}
          numberOfLines={1}
          onLayout={[Function]}
          style={
            {
              "color": "#6200ee",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": [
                {
                  "translateX": 0,
                },
                {
                  "translateY": 0,
                },
                {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Answer here
        </Text>
        <Text
          collapsable={false}
          maxFontSizeMultiplier={1.5}
          numberOfLines={1}
          style={
            {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": [
                {
                  "translateX": 0,
                },
                {
                  "translateY": 0,
                },
                {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Answer here
        </Text>
      </View>
      <TextInput
        editable={true}
        maxFontSizeMultiplier={1.5}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=" "
        placeholderTextColor="rgba(0, 0, 0, 0.54)"
        selectionColor="#6200ee"
        style={
          [
            {
              "flexGrow": 1,
              "margin": 0,
            },
            {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            {
              "height": 64,
            },
            {
              "paddingBottom": 4,
              "paddingTop": 24,
            },
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontVariant": undefined,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            [
              {},
            ],
          ]
        }
        testID="text-input-flat"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>,
  <View
    style={
      [
        {
          "alignItems": "center",
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#40BA62",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "height": 45,
          "justifyContent": "center",
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
          "width": "50%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "disabled": false,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                {
                  "textAlign": "left",
                },
                [
                  {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  {
                    "textTransform": "uppercase",
                  },
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  {
                    "color": "#fff",
                    "fontSize": 18,
                  },
                ],
              ]
            }
          >
            答え合わせ
          </Text>
        </View>
      </View>
    </View>
  </View>,
]
`;

exports[`Quiz tests renders Quiz component 1`] = `
[
  <Text
    style={
      [
        {
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 8,
          "marginLeft": 8,
          "marginRight": 8,
          "marginTop": 8,
          "textAlign": "center",
        },
      ]
    }
  >
    Please write this in 
    Hiroshima
    :
  </Text>,
  <Text
    style={
      [
        {
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 8,
          "marginLeft": 8,
          "marginRight": 8,
          "marginTop": 8,
          "textAlign": "center",
        },
      ]
    }
  >
    Q1
    : 
    あの子は私の知り合いなんだよ。
  </Text>,
  <View
    style={
      [
        {
          "backgroundColor": "rgb(231, 231, 231)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        {
          "margin": 20,
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.26)",
          "bottom": 0,
          "height": 2,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": [
            {
              "scaleY": 0.5,
            },
          ],
          "zIndex": 1,
        }
      }
    />
    <View
      style={
        [
          {
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          {
            "minHeight": 64,
          },
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="none"
        style={
          {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": [
              {
                "translateX": 0,
              },
            ],
            "zIndex": 3,
          }
        }
      >
        <Text
          collapsable={false}
          maxFontSizeMultiplier={1.5}
          numberOfLines={1}
          onLayout={[Function]}
          style={
            {
              "color": "#6200ee",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": [
                {
                  "translateX": 0,
                },
                {
                  "translateY": 0,
                },
                {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Answer here
        </Text>
        <Text
          collapsable={false}
          maxFontSizeMultiplier={1.5}
          numberOfLines={1}
          style={
            {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": [
                {
                  "translateX": 0,
                },
                {
                  "translateY": 0,
                },
                {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Answer here
        </Text>
      </View>
      <TextInput
        editable={true}
        maxFontSizeMultiplier={1.5}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=" "
        placeholderTextColor="rgba(0, 0, 0, 0.54)"
        selectionColor="#6200ee"
        style={
          [
            {
              "flexGrow": 1,
              "margin": 0,
            },
            {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            {
              "height": 64,
            },
            {
              "paddingBottom": 4,
              "paddingTop": 24,
            },
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontVariant": undefined,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            [
              {},
            ],
          ]
        }
        testID="text-input-flat"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>,
  <View
    style={
      [
        {
          "alignItems": "center",
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#40BA62",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "height": 45,
          "justifyContent": "center",
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
          "width": "50%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "disabled": false,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                {
                  "textAlign": "left",
                },
                [
                  {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  {
                    "textTransform": "uppercase",
                  },
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  {
                    "color": "#fff",
                    "fontSize": 18,
                  },
                ],
              ]
            }
          >
            答え合わせ
          </Text>
        </View>
      </View>
    </View>
  </View>,
]
`;

exports[`Quiz tests should contains the heading 1 1`] = `
[
  <Text
    style={
      [
        {
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 8,
          "marginLeft": 8,
          "marginRight": 8,
          "marginTop": 8,
          "textAlign": "center",
        },
      ]
    }
  >
    Please write this in 
    Hiroshima
    :
  </Text>,
  <Text
    style={
      [
        {
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 8,
          "marginLeft": 8,
          "marginRight": 8,
          "marginTop": 8,
          "textAlign": "center",
        },
      ]
    }
  >
    Q1
    : 
    あの子は私の知り合いなんだよ。
  </Text>,
  <View
    style={
      [
        {
          "backgroundColor": "rgb(231, 231, 231)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        {
          "margin": 20,
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.26)",
          "bottom": 0,
          "height": 2,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": [
            {
              "scaleY": 0.5,
            },
          ],
          "zIndex": 1,
        }
      }
    />
    <View
      style={
        [
          {
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          {
            "minHeight": 64,
          },
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="none"
        style={
          {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": [
              {
                "translateX": 0,
              },
            ],
            "zIndex": 3,
          }
        }
      >
        <Text
          collapsable={false}
          maxFontSizeMultiplier={1.5}
          numberOfLines={1}
          onLayout={[Function]}
          style={
            {
              "color": "#6200ee",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": [
                {
                  "translateX": 0,
                },
                {
                  "translateY": 0,
                },
                {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Answer here
        </Text>
        <Text
          collapsable={false}
          maxFontSizeMultiplier={1.5}
          numberOfLines={1}
          style={
            {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": [
                {
                  "translateX": 0,
                },
                {
                  "translateY": 0,
                },
                {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Answer here
        </Text>
      </View>
      <TextInput
        editable={true}
        maxFontSizeMultiplier={1.5}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=" "
        placeholderTextColor="rgba(0, 0, 0, 0.54)"
        selectionColor="#6200ee"
        style={
          [
            {
              "flexGrow": 1,
              "margin": 0,
            },
            {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            {
              "height": 64,
            },
            {
              "paddingBottom": 4,
              "paddingTop": 24,
            },
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontVariant": undefined,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            [
              {},
            ],
          ]
        }
        testID="text-input-flat"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>,
  <View
    style={
      [
        {
          "alignItems": "center",
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#40BA62",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "height": 45,
          "justifyContent": "center",
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
          "width": "50%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "disabled": false,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                {
                  "textAlign": "left",
                },
                [
                  {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  {
                    "textTransform": "uppercase",
                  },
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  {
                    "color": "#fff",
                    "fontSize": 18,
                  },
                ],
              ]
            }
          >
            答え合わせ
          </Text>
        </View>
      </View>
    </View>
  </View>,
]
`;
